ibis/expr/types/generic.py

```diff
@@ -2223,9 +2223,23 @@ class NullColumn(Column, NullValue):
 
 
 @public
-def null():
-    """Create a NULL/NA scalar."""
-    return ops.NULL.to_expr()
+def null(type: dt.DataType | str | None = None) -> Value:
+    """Create a NULL/NA scalar.
+
+    By default, the type will be NULLTYPE. This is castable and comparable to any type,
+    but lacks datatype-specific methods:
+
+    >>> import ibis
+    >>> ibis.null().upper().execute() is None
```

Comment:
```suggestion
    >>> ibis.null().upper()
```

The `execute` here is unnecessary.